### PR TITLE
SEO Audit

### PR DESCRIPTION
**Pull request by mheap**
### Description

Making changes based on the recent SEO audit. This PR contains technical changes (SEO plugin logic) and content changes.

I recommend checking each commit one by one

### Testing instructions

Netlify link: 

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

**Comment by mheap** [comment]
Vale is failing (expected), but all other tests pass

File: app/_hub/kong-inc/acl/_index.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ description: |
   [authentication plugin](/hub/#authentication) (such as
   [Basic Authentication](/hub/kong-inc/basic-auth/),
   [Key Authentication](/hub/kong-inc/key-auth/), [OAuth 2.0](/hub/kong-inc/oauth2/),
-  and [OpenID Connect](/hub/kong-inc/openid-connect/))
+  and [OpenID Connect](/hub/kong-inc/openid-connect/)/)
```
Comment (fabianrbz) [inline app/_hub/kong-inc/acl/_index.md:11]: looks like it doesn't handle the case when there is a ")" right at the end of the link very well :(

**Comment by mheap** [comment]
Broken links grouped by frequency:

```
{
  'http://localhost:8888/gateway/latest/kong-enterprise/security/secrets-management/reference-format/': 8,
  'https://github.com/Kong/kubernetes-ingress-controller/blob/main/docs/faq.md': 8,
  'http://localhost:8888/gateway/kong-enterprise/security/secrets-management/reference-format/': 8,
  'http://localhost:8888/api/vitals.yaml/': 6,
  'https://kubernetes.io/docs/user-guide/kubectl/kubectl_proxy/': 6,
  'http://kubernetes.io/docs/user-guide/service-accounts/': 6,
  'http://kubernetes.io/docs/admin/service-accounts-admin/': 6,
  'http://localhost:8888/enterprise/latest/getting-started': 6,
  'http://localhost:8888/enterprise/1.5.x/deployment/migrations/': 6,
  'https://github.com/Kong/kong-build-tools/tree/master/openresty-build-tools/openresty-patches': 6,
  'http://localhost:8888/code-snippets/spec-renderer.html/': 5,
  'https://github.com/Kong/kubernetes-ingress-controller/blob/main/docs/concepts/design.md': 4,
  'https://github.com/Kong/kubernetes-ingress-controller/blob/main/docs/concepts/deployment.md': 4,
  'https://github.com/Kong/kubernetes-ingress-controller/blob/main/docs/references/version-compatibility.md': 4,
  'https://github.com/Kong/kubernetes-ingress-controller/blob/main/docs/references/cli-arguments.md': 4,
  'https://github.com/Kong/kubernetes-ingress-controller/blob/main/docs/references/plugin-compatibility.md#kong-enterprise': 4,
  'https://github.com/Kong/kubernetes-ingress-controller/blob/main/docs/concepts/ha-and-scaling.md': 4,
  'https://github.com/Kong/kubernetes-ingress-controller/blob/main/docs/concepts/security.md': 4,
  'https://github.com/Kong/kubernetes-ingress-controller/blob/main/docs/deployment': 4,
  'https://github.com/Kong/kubernetes-ingress-controller/blob/main/docs/troubleshooting.md': 4,
  'https://github.com/Kong/kubernetes-ingress-controller/blob/main/docs/concepts/custom-resources.md': 4,
  'https://github.com/Kong/kubernetes-ingress-controller/blob/main/docs/references/annotations.md': 4,
  'https://github.com/Kong/kubernetes-ingress-controller/blob/main/docs/guides': 4,
  'https://github.com/Kong/kubernetes-ingress-controller/blob/main/docs/references/custom-resources.md': 4,
  'https://docs.konghq.com/0.14.x/configuration/': 4,
  'https://getkong.org/docs/0.10.x/proxy': 4,
  'http://localhost:8888/install/redhat/': 4,
  'http://localhost:8888/install/debian/': 4,
  'http://localhost:8888/install/aws-linux/': 4,
  'http://localhost:8888/install/centos/': 4,
  'http://localhost:8888/install/ubuntu/': 4,
  'http://localhost:8888/gateway/key-concepts/routes/': 3,
  'http://localhost:8888/assets/images/diagrams/gslides/kuma_multizone.svg': 3,
  'http://localhost:8888/enterprise/1.3-x/deployment/migrations/': 3,
  'http://localhost:8888/gateway/2.6.x/vitals/vitalsSpec.yaml/': 3,
  'http://localhost:8888/gateway/2.7.x/vitals/vitalsSpec.yaml/': 3,
  'http://localhost:8888/gateway/2.8.x/vitals/vitalsSpec.yaml/': 3,
  'https://cloud.konghq.com/servicehub/': 3,
  'http://localhost:8888/konnect/vitalsSpec.yaml/': 3,
  'http://localhost:8888/gateway/3.0.x/install/linux/os-support/': 2,
  'http://localhost:8888/gateway/3.1.x/install/linux/os-support/': 2,
  'http://localhost:8888/gateway/3.2.x/install/linux/os-support/': 2,
  'https://docs.konghq.com/1.0.x/admin-api': 2,
  'https://docs.konghq.com/0.14.x/health-checks-circuit-breakers/': 2,
  'https://getkong.org/docs/0.13.x/proxy/': 2,
  'https://getkong.org/docs/0.13.x/configuration/': 2,
  'https://getkong.org/docs/0.9.x/configuration/#dns-resolver-section': 2,
  'https://getkong.org/docs/0.6.x/clustering/': 2,
  'https://getkong.org/docs/0.11.x/plugin-development/custom-logic/': 2,
  'https://docs.konghq.com/0.14.x/admin-api/#add-sni': 2,
  'https://docs.konghq.com/0.14.x/admin-api/#add-certificate': 2,
  'https://getkong.org/docs/0.11.x/plugin-development/entities-cache/': 2,
  'https://getkong.org/docs/0.13.x/admin-api/': 2,
  'https://getkong.org/install/docker/': 2,
  'https://okta.com/login/': 2,
  'https://konnect.konghq.com/': 2,
  'http://localhost:8888/install/source': 1,
  'http://localhost:8888/gateway/latest/kong-enterprise/security/secrets-management/getting-started': 1,
  'http://localhost:8888/assets/images/docs/openid-connect/keycloak.json/': 1,
  'https://documentation.pingidentity.com/pingfederate/': 1,
  'http://localhost:8888/kubernetes-ingress-controller/2.8.x/guides/routing-by-header/': 1,
  'https://download.konghq.com/mesh-alpine/kong-mesh-1.8.6-centos-amd64.tar.gz': 1,
  'http://localhost:8888/contributing/terms/': 1,
  'https://github.com/Kong/docs.konghq.com/branches/yours': 1,
  'https://kongponents.netlify.app/style-guide/colors.html': 1,
  'https://github.com/Kong/kubernetes-ingress-controller/blob/main/docs/guides/getting-started.md': 1,
  'https://docs.konghq.com/enterprise/1.5.x/property-reference/#db_cache_warmup_entities': 1,
  'https://docs.konghq.com/1.5.x/cli/#kong-restart': 1,
  'http://localhost:8888/enterprise/1.3-x/developer-portal/overview/#whats-new-in-1.3': 1,
  'http://localhost:8888/enterprise/1.3-x/developer-portal/overview/': 1,
  'http://localhost:8888/enterprise/2.1.x/vitals/vitalsSpec.yaml/': 1,
  'http://localhost:8888/enterprise/2.2.x/vitals/vitalsSpec.yaml/': 1,
  'http://localhost:8888/enterprise/2.3.x/vitals/vitalsSpec.yaml/': 1,
  'http://localhost:8888/gateway/2.4.x/install-and-run/docker/': 1,
  'http://localhost:8888/gateway/2.4.x/install-and-run/docker': 1,
  'http://localhost:8888/enterprise/2.4.x/vitals/vitalsSpec.yaml/': 1,
  'http://localhost:8888/gateway/2.5.x/install-and-run/docker/': 1,
  'http://localhost:8888/gateway/2.5.x/install-and-run/docker': 1,
  'http://localhost:8888/enterprise/2.5.x/vitals/vitalsSpec.yaml/': 1,
  'https://docs.konghq.com/0.15/admin-api': 1,
  'http://localhost:8888/gateway/2.6.x/configure/auth/kong-manager/oidc-mapping/': 1,
  'https://download.konghq.com/gateway-2.x-centos-8/Packages/k/kong-2.6.1.el8.amd64.rpm': 1,
  'http://localhost:8888/gateway/2.6.x/production/running-kong/kong-user/': 1,
  'https://download.konghq.com/gateway-2.x-rhel-8/Packages/k/kong-2.6.1.rhel8.amd64.rpm': 1,
  'https://download.konghq.com/gateway-2.x-centos-8/Packages/k/kong-2.7.2.el8.amd64.rpm': 1,
  'https://download.konghq.com/gateway-2.x-rhel-8/Packages/k/kong-2.7.2.rhel8.amd64.rpm': 1,
  'http://localhost:8888/gateway/admin-api/#vaults-object': 1,
  'http://localhost:8888/gateway/kong-enterprise/security/secrets-management/getting-started/': 1,
  'http://localhost:8888/enterprise/0.33-x/rbac/overview': 1,
  'http://localhost:8888/enterprise/0.33-x/workspaces/overview': 1,
  'http://localhost:8888/enterprise/0.33-x/plugins/statsd-advanced': 1,
  'http://localhost:8888/enterprise/0.36-x/property-reference/#db_cache_warmup_entities': 1,
  'https://javadoc.io/static/com.github.jwt-scala/jwt-core_3.0.0-RC3/7.1.4/api/pdi/jwt/JwtAlgorithm%24%24RS384%24.html': 1,
  'https://github.com/Kong/kong-distributions/pull/766': 1,
  'https://github.com/zhangshuaiNB': 1,
  'http://localhost:8888/gateway/admin-api/': 1,
  'http://localhost:8888/gateway/key-concepts/routes/expressions': 1,
  'http://localhost:8888/gateway/reference/router-expressions-language/': 1,
  'https://konnect.konghq.com/configuration/': 1,
  'https://konnect.konghq.com/runtime-manager': 1
}
```

**Comment by mheap** [comment]
Extracted broken files with incorrect trailing slash with grep:

```
node index.js | grep -E "\.\w+/'"
  'http://localhost:8888/api/vitals.yaml/': 6,
  'http://localhost:8888/code-snippets/spec-renderer.html/': 5,
  'http://localhost:8888/gateway/2.6.x/vitals/vitalsSpec.yaml/': 3,
  'http://localhost:8888/gateway/2.7.x/vitals/vitalsSpec.yaml/': 3,
  'http://localhost:8888/gateway/2.8.x/vitals/vitalsSpec.yaml/': 3,
  'http://localhost:8888/konnect/vitalsSpec.yaml/': 3,
  'http://localhost:8888/assets/images/docs/openid-connect/keycloak.json/': 1,
  'http://localhost:8888/enterprise/2.1.x/vitals/vitalsSpec.yaml/': 1,
  'http://localhost:8888/enterprise/2.2.x/vitals/vitalsSpec.yaml/': 1,
  'http://localhost:8888/enterprise/2.3.x/vitals/vitalsSpec.yaml/': 1,
  'http://localhost:8888/enterprise/2.4.x/vitals/vitalsSpec.yaml/': 1,
  'http://localhost:8888/enterprise/2.5.x/vitals/vitalsSpec.yaml/': 1,
```

Pushed fixes for these items.

All remaining broken links are pre-existing as far as I can tell. We should fix them at some point, but I'd like to get this PR merged soon as it touches a lot of files and will cause conflicts the longer it's open

**Review by fabianrbz — APPROVED** [review]
LGTM! let's :shipit: before it has more conflicts

**Pull request merged**